src/extensions/step_handlers/script_handler/src/script_handler.cpp

```diff
@@ -572,56 +606,92 @@ static ADUC_Result ScriptHandler_PerformAction(const std::string& action, const
         Log_Debug("##########\n# ADU-SHELL ARGS:\n##########\n %s", ss.str().c_str());
     }
 
-    exitCode = ADUC_LaunchChildProcess(ADUSHELL_FILE_PATH, aduShellArgs, scriptOutput);
+    if (prepareArgsOnly)
+    {
+        std::stringstream ss;
+        for (const auto& a : aduShellArgs)
+        {
+            char quote = '\"';
+            char whitespace = ' ';
+
+            // NOLINTNEXTLINE(cppcoreguidelines-avoid-magic-numbers,readability-magic-numbers)
+            if (a[0] != '-')
+            {
+                // If the argument contains a quote character, then use the other quote character.
+                if (a.find(quote) != std::string::npos)
+                {
+                    ss << " '" << a << "'";
+                }
+                else
+                {
+                    ss << " \"" << a << "\"";
+                }
+            }
+            else
+            {
+                ss << " " << a;
+            }
+        }
+        results.scriptOutput = ss.str();
+
+        Log_Debug("Prepare arguments only! adu-shell command:\n\n %s", results.scriptOutput.c_str());
+        results.result = { .ResultCode = ADUC_Result_Success, .ExtendedResultCode = 0 };
+        goto done;
+    }
+
+    exitCode = ADUC_LaunchChildProcess(ADUSHELL_FILE_PATH, aduShellArgs, results.scriptOutput);
 
-    if (!scriptOutput.empty())
+    if (!results.scriptOutput.empty())
     {
-        Log_Info(scriptOutput.c_str());
+        Log_Info(results.scriptOutput.c_str());
     }
 
     if (exitCode != 0)
     {
         int extendedCode = ADUC_ERC_SCRIPT_HANDLER_CHILD_PROCESS_FAILURE_EXITCODE(exitCode);
         Log_Error("Script failed (%s), extendedResultCode:0x%X (exitCode:%d)", action.c_str(), extendedCode, exitCode);
-        result = { .ResultCode = ADUC_Result_Failure, .ExtendedResultCode = extendedCode };
+        results.result = { .ResultCode = ADUC_Result_Failure, .ExtendedResultCode = extendedCode };
```

Comment:
use `/* .ResultCode = */ ADUC_ResultFailures` instead.
MSVC compiler requires C++20 for this syntax, which is c99 syntax actually.